crates/hyperdrive-math/src/lp.rs

```diff
@@ -1047,9 +1603,239 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn fuzz_test_calculate_distribute_excess_idle_withdrawal_shares_redeemed() -> Result<()> {
+        let chain = TestChain::new().await?;
+        let alice = chain.alice().await?;
+        let mock = chain.mock_lp_math();
+
+        // Fuzz the rust and solidity implementations against each other.
+        let mut rng = thread_rng();
+
+        for _ in 0..*FAST_FUZZ_RUNS {
+            // Generate random states.
+            let mut present_state = rng.gen::<State>();
+
+            // Make sure maturity times are in the future.
+            let current_block_timestamp = alice.now().await?;
+            present_state.info.long_average_maturity_time += current_block_timestamp;
+            present_state.info.short_average_maturity_time += current_block_timestamp;
+
+            // active_lp_total_supply and _withdrawal_shares_total_supply are just the token supplies.
+            // Neither are supplied in the PoolInfo so we need to make them here.  lp_total_supply is defined as:
+            // lp_total_supply = active_lp_total_supply + withdrawal_shares_total_supply - withdrawal_shares_ready_to_withdraw
+            // active_lp_total_supply = lp_total_supply - withdrawal_shares_total_supply + withdrawal_shares_ready_to_withdraw
+            // We clip withdrawal_shares_total_supply to ensure active_lp_total_supply doesn't underflow.
+            let active_lp_total_supply = present_state.lp_total_supply()
+                + present_state.withdrawal_shares_ready_to_withdraw();
+            let withdrawal_shares_total_supply = rng.gen_range(fixed!(0)..=active_lp_total_supply);
+            let active_lp_total_supply = active_lp_total_supply - withdrawal_shares_total_supply;
+            // This errors out a lot so we need to catch that here.
+            let starting_present_value =
+                match present_state.calculate_present_value(U256::from(current_block_timestamp)) {
+                    Ok(result) => result,
+                    Err(_) => continue,
```

Comment:
one thing I've done in the past is keep track of a counter of fuzz runs and assert that it is > 0 at the end of the `for` loop. This is probably overkill, but technically `calculate_present_value` could fail every time and the test would pass. 

e.g. here: https://github.com/delvtech/hyperdrive-rs/blob/main/crates/hyperdrive-math/src/short/open.rs#L653-L703

Reply:
Yeah I do that too while writing the tests to make sure that I'm seeing success cases.  I dunno that we want it in  everywhere though because it might cause flakes

Reply:
I just checked and most of these are hitting success over 1/2 the time so im gonna leave it out